goaviatrix/controller_bgp_max_as_limit_config.go

```diff
@@ -0,0 +1,70 @@
+package goaviatrix
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func (c *Client) SetControllerBgpMaxAsLimit(ctx context.Context, maxAsLimit int) error {
+	data := map[string]string{
+		"action":       "set_bgp_max_as_limit",
+		"CID":          c.CID,
+		"max_as_limit": fmt.Sprint(maxAsLimit),
+	}
+
+	checkFunc := func(action, reason string, ret bool) error {
+		if !ret && !strings.Contains(reason, "Configured BGP maximum AS limit is not changed") {
+			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
+		}
+		return nil
+	}
+	return c.PostAPIContext(ctx, data["action"], data, checkFunc)
+}
+
+func (c *Client) DisableControllerBgpMaxAsLimit(ctx context.Context) error {
+	data := map[string]string{
+		"action":       "set_bgp_max_as_limit",
+		"CID":          c.CID,
+		"max_as_limit": "",
+	}
+
+	checkFunc := func(action, reason string, ret bool) error {
+		if !ret && !strings.Contains(reason, "Configured BGP maximum AS limit is not changed") {
+			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
+		}
+		return nil
+	}
+	return c.PostAPIContext(ctx, data["action"], data, checkFunc)
+}
+
+func (c *Client) GetControllerBgpMaxAsLimit(ctx context.Context) (int, error) {
+	data := map[string]string{
+		"action": "show_bgp_max_as_limit",
+		"CID":    c.CID,
+	}
+
+	type BgpMaxAsLimitResponse struct {
+		Results string
+	}
+
+	var resp BgpMaxAsLimitResponse
+	err := c.GetAPIContext(ctx, &resp, data["action"], data, BasicCheck)
+	if err != nil {
+		return 0, err
+	}
+
+	if resp.Results == "" {
+		return -1, nil
```

Comment:
Don't use -1 as some special value, communicate what exactly went wrong through the `error` not the value. In this case I think you would return ErrNotFound here

Reply:
I'm using -1 to signify that the feature is currently disabled. This is for an edge case where the user created the resource and somehow disabled the feature somewhere else (maybe through the UI). If I return an error it wouldn't be possible for them to update the resource. They would have to destroy it first then create a new one.

Reply:
You can use a specific error to communicate this case, like ErrNotFound, in you Read function you would check for ErrNotFound.

user would not need to destroy the resource first, in the Read if you d.SetID(“”) then terraform considers the resource already destroyed